rs/rosetta-api/icrc1/rosetta/client/src/lib.rs

```diff
@@ -180,20 +180,11 @@ impl RosettaClient {
         while tries < 10 {
             let transaction = self
                 .search_transactions(
-                    network_identifier.clone(),
-                    Some(submit_response.transaction_identifier.clone()),
-                    None,
-                    None,
-                    None,
-                    None,
-                    None,
+                    &SearchTransactionsRequest::builder(network_identifier.clone())
+                        .with_transaction_identifier(submit_response.transaction_identifier.clone())
+                        .build(),
```

Comment:
I like the use of a builder here! Could this `SearchTransactionsRequest` instance be created outside the `while` loop, since only a non-mutable reference is required to be passed to `self.search_transactions()`?